### ISSUE TITLE
2.1.0 release

### ISSUE THREAD

**Issue by jorisvandenbossche** [shapely/shapely#2144]
Issue to track and plan the 2.1.0 release (which is long overdue ..).

I would like to do a 2.1.0 release in one of the coming weeks. Besides some remaining feature PRs that would be nice to get in, I think we have two others questions:

- For the wheels, do we already bump the included GEOS version to the just-released 3.13.0 release, or do stick with the 3.11->3.12 update? 
- Do we drop Python 3.9 support before or after doing the 2.1.0 release? (https://github.com/shapely/shapely/issues/2119)

We don't use the milestone consistently, but nevertheless the issues currently tagged with the 2.1 milestone: https://github.com/shapely/shapely/milestone/52

cc @shapely/shapely-team

**Comment by jorisvandenbossche** [comment]
In terms of remaining features to include, ideally we merge as many of the open PRs as possible, but I would say without holding up the release if not ready. The two bigger features I would definitely like to include:

- [ ] https://github.com/shapely/shapely/pull/1969
- [ ] https://github.com/shapely/shapely/pull/1307

I also just opened an API question that might be good to discuss/resolve:

- [ ] https://github.com/shapely/shapely/issues/2143

**Comment by theroggy** [comment]
In an ideal world, I think it would be great if `GEOSCoverageIsValid` would become available together with https://github.com/shapely/shapely/pull/1969, as I'm afraid otherwise there might be quite some cases where coverage simplify doesn't produce expected results and it is not possible/difficult to check if the coverage not being a valid coverage is the reason.

**Comment by jorisvandenbossche** [comment]
> In an ideal world, I think it would be great if `coverage_is_valid` would become available together with `coverage_simplify` (#1969).

-> https://github.com/shapely/shapely/pull/2156

**Comment by mwtoews** [comment]
My main goals for a future release is #1648 which is partially done for 2.1.x but not complete, and won't make it for this release time-line. However, I'm wondering about adding some warnings about future breaking changes. A few of these breaking ideas are roughly drafted in https://github.com/shapely/shapely/issues/1648#issuecomment-1504168649.

**Comment by sgillies** [comment]
I don't have particular features in mind, am 👍 on bringing functions in from the latest GEOS.

**Comment by theroggy** [comment]
GEOS version 3.13.0 isn't very new anymore by now... so I'd skip 3.12 for the shapely 2.1 wheels... There is already a PR for the upgrade as well: #2187

**Comment by rcomer** [comment]
GEOS v3.13 fixed a bug that has been plaguing Cartopy users for a long time, so it would be great for Cartopy if the next Shapely wheels included it.

**Comment by jorisvandenbossche** [comment]
> GEOS version 3.13.0 isn't very new anymore by now... so I'd skip 3.12 for the shapely 2.1 wheels... There is already a PR for the upgrade as well: [#2187](https://github.com/shapely/shapely/pull/2187)

+1, will merge that PR

**Comment by jorisvandenbossche** [comment]
Shall we already do a 2.1.0rc0 to get things rolling (and let people test the upcoming version more easily)

**Comment by sgillies** [comment]
@jorisvandenbossche how about a 2.1a1, since we have 2.1.0 features still in development. I think people generally don't expect API additions in a second release candidate.

**Comment by Zaczero** [comment]
@jorisvandenbossche That would be great if possible. I've been installing shapely from git but it doesn't seem to work with build isolation (PEP 518) and the experience was not great. I've already started testing of the extended make_valid functionality 😜.

**Comment by jorisvandenbossche** [comment]
@Zaczero if you want to easily test the development version without having to build yourself, you should also be able to install the nightly wheel with:

```
 pip install -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple shapely 
```

**Comment by jorisvandenbossche** [comment]
> how about a 2.1a1, since we have 2.1.0 features still in development: https://github.com/shapely/shapely/milestone/52. I think people generally don't expect API additions in a second release candidate. 

@sgillies Yeah, but then the idea of doing a release candidate was then indeed to stop adding new features and just some final fixes. The milestone indeed has some features listed, but mostly things that are essentially ready (I would merge the PRs for orient_polygons and coverage_is_valid before doing the RC).  
I think the main ongoing work is the M support by @mwtoews and so it depends a bit on how much we want to get in for that (https://github.com/shapely/shapely/issues/1648)

**Comment by mwtoews** [comment]
I have a few deprecated positional arguments to bring into 2.1 #2237, but after that I'm keen seeing a 2.1a1 release.

I'll also look into getting a GEOS 3.13.1 ready for the binary wheels, mostly to resolve `repr()` versions of geometries with quirks related to https://github.com/libgeos/geos/issues/1198

**Comment by theroggy** [comment]
https://github.com/shapely/shapely/pull/1685 is also a candidate I think to include in 2.1, the only open question there is the name of the function I think...

**Comment by jorisvandenbossche** [comment]
I am tentatively planning to do a 2.1.0rc0 release tomorrow (and then hopefully a final 2.1.0 release in 1-2 weeks if all goes well).

I think all critical PRs for the release have been merged (https://github.com/shapely/shapely/milestone/52), and I have a PR to update the release notes in https://github.com/shapely/shapely/pull/2254.

**Comment by jorisvandenbossche** [comment]
Release candidate is out: https://github.com/shapely/shapely/releases/tag/2.1.0rc1 / https://pypi.org/project/shapely/2.1.0rc1/

**Comment by jorisvandenbossche** [comment]
It has been two weeks without any issue, so I would propose I do the 2.1.0 release tomorrow.

**Comment by jorisvandenbossche** [comment]
Shapely 2.1.0 is released! And up on PyPI and conda-forge. 
https://shapely.readthedocs.io/en/latest/release/2.x.html#version-2-1-0

**Comment by jorisvandenbossche** [comment]
Opened a follow-up issue for 2.1.x patch releases (https://github.com/shapely/shapely/issues/2292), and closing this one
